client/utils.py

```diff
@@ -148,6 +149,33 @@ def write_command(fdesc, command, destinations, source=0):
     time.sleep(0.3)
 
 
+def write_command_retry(fdesc, command, destinations, source=0):
+    """
+    Writes a command, retries as long as there is no answer and returns a dictionnary containing
+    a map of each board ID and its answer.
+    """
+    write_command(fdesc, command, destinations, source)
+    reader = CANDatagramReader(fdesc)
+    answers = dict()
+
+    while len(answers) < len(destinations):
```

Comment:
I agree on your first point, but not on your second. I would prefer the update operation to only pass or fail. I don't want half the robot running the new version and half the robot running the old one.